psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -1356,23 +1356,78 @@ std::vector<std::vector<std::tuple<double, int, int>>> Wavefunction::get_no_occu
     return no_occs;
 }
 
-double Wavefunction::get_variable(std::string label) {
-    std::string uc_label = label;
 
-    if (variables_.count(uc_label) == 0) {
-        throw PSIEXCEPTION("Wavefunction::get_variable: Requested variable " + label + " was not set!\n");
+int Wavefunction::has_scalar_variable(const std::string &key) {
+    std::string uc_key = key;
+    std::transform(uc_key.begin(), uc_key.end(), uc_key.begin(), ::toupper);
```

Comment:
`psi4/libpsi4util/stl_string.cc` has a `to_upper_copy` function, so there's no need to do it "by hand" here.

Reply:
Thanks, this solved a mystery for me.. I had tried to add a `to_upper` fn into `wavefunction.cc` last week, and it kept complaining about bad `const`. It mush have been accessing the `stl_sting.cc` one. Since your cmake renovation, libpsi4util isn't accessible, hence the added header.